libsolidity/analysis/DocStringTagParser.cpp

```diff
@@ -199,40 +210,41 @@ void DocStringTagParser::parseDocStrings(
 		else if (tagName == "return")
 		{
 			returnTagsVisited++;
+
+			vector<string> returnParameterNames;
+
 			if (auto const* varDecl = dynamic_cast<VariableDeclaration const*>(&_node))
 			{
 				solAssert(varDecl->isPublic(), "@return is only allowed on public state-variables.");
-				if (returnTagsVisited > 1)
-					m_errorReporter.docstringParsingError(
-						5256_error,
-						_node.documentation()->location(),
-						"Documentation tag \"@" + tagName + "\" is only allowed once on state-variables."
-					);
+				returnParameterNames = FunctionType(*varDecl).returnParameterNames();
 			}
 			else if (auto const* function = dynamic_cast<FunctionDefinition const*>(&_node))
-			{
-				string content = tagValue.content;
-				string firstWord = content.substr(0, content.find_first_of(" \t"));
+				returnParameterNames = FunctionType(*function).returnParameterNames();
+			else
+				continue;
+
+			string content = tagValue.content;
+			string firstWord = content.substr(0, content.find_first_of(" \t"));
 
-				if (returnTagsVisited > function->returnParameters().size())
+			if (returnTagsVisited > returnParameterNames.size())
+				m_errorReporter.docstringParsingError(
+					2604_error,
```

Comment:
This error is semantically pretty close to the old one. Maybe it would be better to preserve the old error code?
```suggestion
					5256_error,
```

Reply:
I did not introduce any new errors nor changed any old ones, there only has been some moving around..

Reply:
Right. The diff is a bit hard to read here and I thought `2604_error` was being replaced by `5856_error` but it's still there. It's only 5256_error` that is being removed.

Reply:
Nope, `5256` is still here, too